MediaBrowser.MediaEncoding/Probing/FFProbeHelpers.cs

```diff
@@ -57,6 +57,90 @@ public static string GetDictionaryValue(IReadOnlyDictionary<string, string> tags
+        /// Gets a string from an FFProbeResult tags dictionary, checking all keys sequentially,
+        /// stopping at the first key that returns a result that's neither null nor blank.
+        /// </summary>
+        /// <param name="tags">The tags.</param>
+        /// <param name="key1">The first checked key.</param>
+        /// <returns>System.String.</returns>
+        public static string GetFirstNonNullNorBlankDictionaryValue(IReadOnlyDictionary<string, string> tags, string key1)
+        {
+            if (tags == null)
+            {
+                return null;
+            }
+
+            if (tags.TryGetValue(key1, out var val) && !string.IsNullOrWhiteSpace(val))
+            {
+                return val;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Gets a string from an FFProbeResult tags dictionary, checking all keys sequentially,
+        /// stopping at the first key that returns a result that's neither null nor blank.
+        /// </summary>
+        /// <param name="tags">The tags.</param>
+        /// <param name="key1">The first checked key.</param>
+        /// <param name="key2">The second checked key.</param>
+        /// <returns>System.String.</returns>
+        public static string GetFirstNonNullNorBlankDictionaryValue(IReadOnlyDictionary<string, string> tags, string key1, string key2)
+        {
+            if (tags == null)
+            {
+                return null;
+            }
+
+            if (tags.TryGetValue(key1, out var val) && !string.IsNullOrWhiteSpace(val))
+            {
+                return val;
+            }
+
+            if (tags.TryGetValue(key2, out val) && !string.IsNullOrWhiteSpace(val))
+            {
+                return val;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Gets a string from an FFProbeResult tags dictionary, checking all keys sequentially,
+        /// stopping at the first key that returns a result that's neither null nor blank.
+        /// </summary>
+        /// <param name="tags">The tags.</param>
+        /// <param name="key1">The first checked key.</param>
+        /// <param name="key2">The second checked key.</param>
+        /// <param name="key3">The third checked key.</param>
+        /// <returns>System.String.</returns>
+        public static string GetFirstNonNullNorBlankDictionaryValue(IReadOnlyDictionary<string, string> tags, string key1, string key2, string key3)
+        {
+            if (tags == null)
+            {
+                return null;
+            }
+
+            if (tags.TryGetValue(key1, out var val) && !string.IsNullOrWhiteSpace(val))
+            {
+                return val;
+            }
+
+            if (tags.TryGetValue(key2, out val) && !string.IsNullOrWhiteSpace(val))
+            {
+                return val;
+            }
+
+            if (tags.TryGetValue(key3, out val) && !string.IsNullOrWhiteSpace(val))
```

Comment:
"a lot" of code duplication here. I think it would be better to create one method that handles 3 keys and simply sending string.Empty from the other two methods.

Eg. `if (!string.IsNullOrEmpty(key3) && tags.TryGetValue(key3, out val) && !string.IsNullOrWhiteSpace(val))`

Reply:
Right, copy pasting those wasn't the greatest decision. And I think so, yes, but the other method should be moved as well then.

Reply:
> I think so, yes, but the other method should be moved as well then.

Do you mean GetDictionaryValue? There is already an extension method for this:
[GetValueOrDefault](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.collectionextensions.getvalueordefault?view=net-5.0#System_Collections_Generic_CollectionExtensions_GetValueOrDefault__2_System_Collections_Generic_IReadOnlyDictionary___0___1____0_)

Reply:
> Do you mean GetDictionaryValue? There is already an extension method for this:
> [GetValueOrDefault](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.collectionextensions.getvalueordefault?view=net-5.0#System_Collections_Generic_CollectionExtensions_GetValueOrDefault__2_System_Collections_Generic_IReadOnlyDictionary___0___1____0_). The other methods could be moved though

Addressed this now.